src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private static OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = new HashMap<>();

```

Comment:
The `storage` map should not be part of the `ShopService` implementation. Consider abstracting the storage functionality into a separate class and using dependency injection to pass it to services that need it. This will help you adhere to the Single Responsibility Principle and make your code more modular.